util/deadlockhistory/deadlock_history_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deadlockhistory
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/kvproto/pkg/deadlock"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	tikverr "github.com/pingcap/tidb/store/tikv/error"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/resourcegrouptag"
+)
+
+type testDeadlockHistorySuite struct{}
+
+var _ = Suite(&testDeadlockHistorySuite{})
+
+func TestT(t *testing.T) {
+	TestingT(t)
+}
+
+func (s *testDeadlockHistorySuite) TestDeadlockHistoryCollection(c *C) {
```

Comment:
How about add some field assertion for the `head` and `size` especially for the over capacity path?

Reply:
done